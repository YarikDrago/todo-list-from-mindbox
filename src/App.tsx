import React, {useState} from 'react';
import './App.css';
import styled from 'styled-components'
import ListButton from "./components/ListButton";
import ListInput from "./components/ListInput";
import TaskSchedule from "./components/TaskSchedule";
import ListData from "./ListData";
import {observer} from "mobx-react";
import MainTitle from "./components/MainTitle/MainTitle";


const StyledApp = styled.article`
  display: flex;
  //justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100vw;
  height: 100vh;
  
  background-color: #f2f2f2;
`

const TitleTodos = styled.h1`
  //font-family: 'Comfortaa', cursive;
  font-family: 'Yantramanav', sans-serif;
  font-weight: 100;  
  color: #e7d7d6;
  font-size: 7rem;
  margin: 0;
`

const ToDoMainForm = styled.main`
  display: block;
  box-sizing: border-box;
  width: 600px;
  height: auto;
  //padding: 10px;  
  background-color: #fcfcfc;
  box-shadow: rgba(100, 100, 111, 0.5) 5px 7px 29px 0;
`

const List = styled.main`
  //box-shadow: rgba(100, 100, 111, 0.5) 5px 7px 29px 0;
`

const ListFooter = styled.footer`
  display: flex;
  align-items: center;
  justify-content: space-between;  
`


const App = observer(()=> {

    const [activeBtn, setActiveBtn] = useState("all");

    function showAll(){
        setActiveBtn("all")
        ListData.changeShowOption("all")
        // console.log("show all")
    }

    function showActive(){
        setActiveBtn("active")
        ListData.changeShowOption("active")
        // console.log("show active")
    }

    function showComleted(){
        setActiveBtn("completed")
        ListData.changeShowOption("completed")
        // console.log("show completed")
    }

    function clearCompleted(){
        console.log("clear completed")
        ListData.clearCompleted()
    }

  return (
    <StyledApp className="App">
        <MainTitle>todos</MainTitle>
        <ToDoMainForm>
            <ListInput/>
            <TaskSchedule/>
            <ListFooter>
                <p style={{marginLeft: "10px"}}>{`${ListData.incomletedTasks} items left`}</p>
                <nav>
                    <ListButton
                        className={activeBtn === "all" ? "active" : ""}
                        onClick={()=>{showAll()}}
                    >All</ListButton>
                    <ListButton
                        className={activeBtn === "active" ? "active" : ""}
                        onClick={()=>{showActive()}}
                    >Active</ListButton>
                    <ListButton
                        className={activeBtn === "completed" ? "active" : ""}
                        onClick={()=>{showComleted()}}
                    >Completed</ListButton>
                </nav>
                <ListButton
                    onClick={()=>{clearCompleted()}}
                >Clear completed</ListButton>
            </ListFooter>
        </ToDoMainForm>

    </StyledApp>
  );
})

export default App;
