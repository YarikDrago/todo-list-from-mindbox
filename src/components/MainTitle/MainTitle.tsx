import React, {ReactElement, useEffect, useRef} from 'react';
import styled from 'styled-components'

const StyledTitle =  styled.h1`
  font-family: 'Yantramanav', sans-serif;
  font-weight: 100;  
  color: #e7d7d6;
  font-size: 7rem;
  margin: 0;
`

const MainTitle = ({...props}) => {
    const refTitle = useRef(null)

    return (
        <StyledTitle
            ref = {refTitle}
            data-testid={'main-title'}
        >
            {props.children}
        </StyledTitle>
    );
};

export default MainTitle;