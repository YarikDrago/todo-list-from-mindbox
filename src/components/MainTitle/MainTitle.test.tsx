import MainTitle from "./MainTitle";
import {render, screen} from "@testing-library/react";
import React from "react";
import App from "../../App";

describe('test of main title', ()=>{
    test('exist of element', ()=>{
        render(<MainTitle/>)
        const mainTitleElem = screen.getByTestId('main-title');
        expect(mainTitleElem).toBeInTheDocument();
    })
    test('check heading text', ()=>{
        render(<App/>)
        const mainTitleElem = screen.getByTestId('main-title');
        expect(mainTitleElem.textContent).toBe('todos')
        expect(mainTitleElem.textContent).toMatchSnapshot()
    })
})