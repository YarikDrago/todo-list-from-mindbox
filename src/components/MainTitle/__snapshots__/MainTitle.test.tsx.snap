// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test of main title check heading text 1`] = `"todos"`;
